server/storage/backend/batch_tx.go

```diff
@@ -310,7 +314,14 @@ func (t *batchTxBuffered) Unlock() {
 		t.buf.writeback(&t.backend.readTx.buf)
 		// gofail: var afterWritebackBuf struct{}
 		t.backend.readTx.Unlock()
-		if t.pending >= t.backend.batchLimit {
+		// We need to commit the transaction if
+		// 1. t.pending >= t.backend.batchLimit for performance,
+		// 	  because We don't want buffer to grow too big as it has worse complexity.
```

Comment:
That's my current understanding, however would be good to confirm. Could you check PRs when the batchlimit was introduced and see what was the motivation back then. Referencing PR would be also useful.

cc @ahrtr

Reply:
this is one relevant pr https://github.com/etcd-io/etcd/pull/10283